### PR TITLE
tap: fix error-prone hardcoding of tap names max length

### PR DESCRIPTION
**Pull request by acatangiu**
## Reason for This PR

Small refactor to make the tap-name code more robust.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.

### CODE REVIEW

File: net_util/src/tap.rs [hunk 1]
```diff
@@ -16,6 +16,8 @@ use libc;
 use net_gen;
 use sys_util::{ioctl_with_mut_ref, ioctl_with_ref, ioctl_with_val};
 
+const IFACE_NAME_MAX_LEN: usize = 16;
```
Comment (andreeaflorescu) [inline net_util/src/tap.rs]: I suggest adding a comment here to say that this value is taken from the Linux kernel: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/if.h#L33

File: net_util/src/tap.rs [hunk 2]
```diff
@@ -68,18 +70,19 @@ fn create_socket() -> Result<UdpSocket> {
...
 // Returns a byte vector representing the contents of a null terminated C string which
 // contains if_name.
-fn build_terminated_if_name(if_name: &str) -> Result<Vec<u8>> {
+fn build_terminated_if_name(if_name: &str) -> Result<[u8; IFACE_NAME_MAX_LEN]> {
     // Convert the string slice to bytes, and shadow the variable,
     // since we no longer need the &str version.
     let if_name = if_name.as_bytes();
 
-    // TODO: the 16usize limit of the if_name member from struct Tap is pretty arbitrary.
+    // TODO: the IFACE_NAME_MAX_LEN (16usize) limit of the if_name member from struct Tap is
```
Comment (andreeaflorescu) [inline net_util/src/tap.rs]: It's not really arbitrary, that's the maxsize defined in the kernel.

Can you also update the commit message to specify that the change is needed to conform with the Linux Kernel definition?

File: net_util/src/tap.rs (hunk 1 above)
Comment (acatangiu) [reply net_util/src/tap.rs]: Sure. I've also added a test that validates this constant is still up to date.

File: net_util/src/tap.rs (hunk 2 above)
Comment (acatangiu) [reply net_util/src/tap.rs]: Removed this comment as it doesn't make much sense any more. Updated the commit message.

**Comment by acatangiu** [comment]
@andreeaflorescu @serban300 comments addressed, please re-review.

**Pull request merged**